Amplitude/Amplitude.m

```diff
@@ -310,17 +320,13 @@ - (void)initializeApiKey:(NSString*) apiKey userId:(NSString*) userId
         return;
     }
 
-    (void) SAFE_ARC_RETAIN(apiKey);
+    SAFE_ARC_RETAIN(apiKey);
     SAFE_ARC_RELEASE(_apiKey);
     _apiKey = apiKey;
     
     [self runOnBackgroundQueue:^{
         @synchronized (_eventsData) {
-            if (userId != nil) {
-                [self setUserId:userId];
-            } else {
-                _userId = SAFE_ARC_RETAIN([_eventsData objectForKey:@"user_id"]);
-            }
+            [self setUserId: setUserId ? userId : [_eventsData objectForKey:@"user_id"]];
```

Comment:
i think this should only get called if setUserId is YES, otherwise it's a noop call that forces an unnecessary saveEventsData call